### PR TITLE
Refactor move core out from android

### PR DESCRIPTION
**Pull request by LarryHsiao**

### CODE REVIEW

**Comment by LarryHsiao** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 10
           

Complexity increasing per file
==============================
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotsByKeyword.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotsByIds.java  5
- core/src/main/java/com/larryhsiao/nyx/core/tags/JotsByTagId.java  3
- core/src/main/java/com/larryhsiao/nyx/core/attachments/NewAttachment.java  5
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotById.java  5
- app/src/main/java/com/larryhsiao/nyx/jot/AttachmentAdapter.java  1
- app/src/main/java/com/larryhsiao/nyx/jot/JotListFragment.java  5
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotsByDate.java  3
- core/src/main/java/com/larryhsiao/nyx/core/attachments/AttachmentDb.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/JotContentFragment.java  8
- app/src/main/java/com/larryhsiao/nyx/tag/TagListAdapter.java  2
- core/src/main/java/com/larryhsiao/nyx/core/tags/QueriedTags.java  4
- core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java  2
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotRemoval.java  3
- app/src/main/java/com/larryhsiao/nyx/base/AuthActivity.java  1
- app/src/main/java/com/larryhsiao/nyx/jot/JotMapItem.java  1
- core/src/main/java/com/larryhsiao/nyx/core/attachments/NewAttachments.java  2
- core/src/main/java/com/larryhsiao/nyx/core/tags/JotTagRemoval.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/AllJots.java  3
- core/src/test/java/com/larryhsiao/nyx/jots/JotUriIdTest.java  2
- core/src/main/java/com/larryhsiao/nyx/core/jots/QueriedJots.java  5
- core/src/main/java/com/larryhsiao/nyx/core/attachments/QueriedAttachments.java  4
- core/src/main/java/com/larryhsiao/nyx/core/attachments/ConstAttachment.java  1
- app/src/main/java/com/larryhsiao/nyx/jot/JotMonthView.java  11
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotsByLocation.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/ConstJot.java  1
- core/src/main/java/com/larryhsiao/nyx/core/jots/NewJot.java  9
- core/src/main/java/com/larryhsiao/nyx/core/tags/NewTag.java  5
- core/src/test/java/com/larryhsiao/nyx/jots/JotsByLocationTest.java  1
- core/src/main/java/com/larryhsiao/nyx/core/tags/TagsByKeyword.java  3
- core/src/test/java/com/larryhsiao/nyx/jots/NewJotTest.java  2
- core/src/main/java/com/larryhsiao/nyx/core/tags/TagDb.java  3
- app/src/main/java/com/larryhsiao/nyx/util/EmptyView.java  1
- core/src/test/java/com/larryhsiao/nyx/jots/QueriedJotsTest.java  1
- core/src/main/java/com/larryhsiao/nyx/core/attachments/AttachmentsByJotId.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/JotWeekView.java  11
- app/src/main/java/com/larryhsiao/nyx/jot/JotListAdapter.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/WrappedJot.java  1
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotUriId.java  3
- core/src/main/java/com/larryhsiao/nyx/core/tags/NewJotTag.java  3
- core/src/test/java/com/larryhsiao/nyx/jots/JotUriTest.java  1
- app/src/main/java/com/larryhsiao/nyx/tag/TagListFragment.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/CalendarFragment.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/JotMapFragment.java  7
- core/src/main/java/com/larryhsiao/nyx/core/tags/TagRemoval.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotsDb.java  3
- core/src/test/java/com/larryhsiao/nyx/attachments/NewAttachmentTest.java  1
- core/src/main/java/com/larryhsiao/nyx/core/tags/AllTags.java  3
- core/src/main/java/com/larryhsiao/nyx/core/tags/ConstTag.java  1
- core/src/main/java/com/larryhsiao/nyx/core/attachments/RemovalAttachmentByJotId.java  3
- app/src/main/java/com/larryhsiao/nyx/LocationString.java  7
- core/src/main/java/com/larryhsiao/nyx/core/tags/TagsByJotId.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/UpdateJot.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotUri.java  1
- core/src/main/java/com/larryhsiao/nyx/core/jots/PostedJot.java  2
- app/src/main/java/com/larryhsiao/nyx/base/JotFragment.java  1
- core/src/main/java/com/larryhsiao/nyx/core/jots/SampleJot.java  1
- app/src/main/java/com/larryhsiao/nyx/base/JotActivity.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/PointSource.java  2
         

Complexity decreasing per file
==============================
+ app/src/main/java/com/larryhsiao/nyx/JotApplication.java  -3
         

Clones added
============
- app/src/main/java/com/larryhsiao/nyx/jot/JotListFragment.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/JotMonthView.java  2
- app/src/main/java/com/larryhsiao/nyx/jot/JotWeekView.java  2
- app/src/main/java/com/larryhsiao/nyx/tag/TagListFragment.java  2
- app/src/main/java/com/larryhsiao/nyx/jot/JotMapFragment.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067&bid=16884172)

File: core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import com.larryhsiao.nyx.core.tags.TagDb;
+import com.silverhetch.clotho.Source;
+import com.silverhetch.clotho.database.h2.EmbedH2Conn;
+import com.silverhetch.clotho.source.ConstSource;
+import org.flywaydb.core.Flyway;
+
+import java.io.File;
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
```
Comment (LarryHsiao) [inline core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java]: Codacy Issue found: [Avoid unused imports such as 'java.sql.ResultSet'](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067)

File: app/src/main/java/com/larryhsiao/nyx/JotApplication.java [hunk 2]
```diff
@@ -1,14 +1,14 @@
 package com.larryhsiao.nyx;
 
 import android.app.Application;
-import com.larryhsiao.nyx.attachments.AttachmentDb;
-import com.larryhsiao.nyx.jots.JotsDb;
-import com.larryhsiao.nyx.tags.TagDb;
+import com.larryhsiao.nyx.core.NyxDb;
+import com.larryhsiao.nyx.core.attachments.AttachmentDb;
```
Comment (LarryHsiao) [inline app/src/main/java/com/larryhsiao/nyx/JotApplication.java]: Codacy Issue found: [Unused import - com.larryhsiao.nyx.core.attachments.AttachmentDb.](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067)

File: app/src/main/java/com/larryhsiao/nyx/JotApplication.java [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
+import com.larryhsiao.nyx.core.jots.JotsDb;
```
Comment (LarryHsiao) [inline app/src/main/java/com/larryhsiao/nyx/JotApplication.java]: Codacy Issue found: [Avoid unused imports such as 'com.larryhsiao.nyx.core.jots.JotsDb'](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067)

Comment (LarryHsiao) [inline app/src/main/java/com/larryhsiao/nyx/JotApplication.java]: Codacy Issue found: [Unused import - com.larryhsiao.nyx.core.jots.JotsDb.](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067)

File: core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import com.larryhsiao.nyx.core.jots.JotsDb;
+import com.larryhsiao.nyx.core.tags.TagDb;
+import com.silverhetch.clotho.Source;
+import com.silverhetch.clotho.database.h2.EmbedH2Conn;
+import com.silverhetch.clotho.source.ConstSource;
+import org.flywaydb.core.Flyway;
+
+import java.io.File;
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
```
Comment (LarryHsiao) [inline core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java]: Codacy Issue found: [Avoid unused imports such as 'java.sql.DatabaseMetaData'](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067)

File: app/src/main/java/com/larryhsiao/nyx/JotApplication.java [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
+import com.larryhsiao.nyx.core.tags.TagDb;
```
Comment (LarryHsiao) [inline app/src/main/java/com/larryhsiao/nyx/JotApplication.java]: Codacy Issue found: [Unused import - com.larryhsiao.nyx.core.tags.TagDb.](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067)

Comment (LarryHsiao) [inline app/src/main/java/com/larryhsiao/nyx/JotApplication.java]: Codacy Issue found: [Avoid unused imports such as 'com.larryhsiao.nyx.core.tags.TagDb'](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067)

File: core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java (hunk 4 above)
Comment (LarryHsiao) [inline core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java]: Codacy Issue found: [Unused import - java.sql.DatabaseMetaData.](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067)

File: core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java (hunk 1 above)
Comment (LarryHsiao) [inline core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java]: Codacy Issue found: [Unused import - java.sql.ResultSet.](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067)

File: app/src/main/java/com/larryhsiao/nyx/JotApplication.java (hunk 2 above)
Comment (LarryHsiao) [inline app/src/main/java/com/larryhsiao/nyx/JotApplication.java]: Codacy Issue found: [Avoid unused imports such as 'com.larryhsiao.nyx.core.attachments.AttachmentDb'](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067)

**Comment by LarryHsiao** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- core/src/main/java/com/larryhsiao/nyx/core/tags/TagsByJotId.java  3
- core/src/main/java/com/larryhsiao/nyx/core/attachments/NewAttachment.java  5
- app/src/main/java/com/larryhsiao/nyx/jot/JotListAdapter.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/SampleJot.java  1
- app/src/main/java/com/larryhsiao/nyx/jot/JotMapItem.java  1
- core/src/main/java/com/larryhsiao/nyx/core/attachments/AttachmentsByJotId.java  3
- app/src/main/java/com/larryhsiao/nyx/util/EmptyView.java  1
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotUriId.java  3
- core/src/main/java/com/larryhsiao/nyx/core/tags/AllTags.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/NewJot.java  9
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotById.java  5
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotsByDate.java  3
- core/src/test/java/com/larryhsiao/nyx/attachments/NewAttachmentTest.java  1
- core/src/main/java/com/larryhsiao/nyx/core/jots/PointSource.java  2
- core/src/main/java/com/larryhsiao/nyx/core/attachments/RemovalAttachmentByJotId.java  3
- core/src/main/java/com/larryhsiao/nyx/core/tags/JotsByTagId.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/JotContentFragment.java  8
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotUri.java  1
- app/src/main/java/com/larryhsiao/nyx/base/AuthActivity.java  1
- core/src/main/java/com/larryhsiao/nyx/core/tags/NewJotTag.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/AllJots.java  3
- core/src/main/java/com/larryhsiao/nyx/core/attachments/QueriedAttachments.java  4
- app/src/main/java/com/larryhsiao/nyx/jot/JotListFragment.java  5
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotRemoval.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/AttachmentAdapter.java  1
- core/src/test/java/com/larryhsiao/nyx/jots/JotUriIdTest.java  2
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotsByKeyword.java  3
- app/src/main/java/com/larryhsiao/nyx/tag/TagListAdapter.java  2
- core/src/main/java/com/larryhsiao/nyx/core/tags/ConstTag.java  1
- core/src/test/java/com/larryhsiao/nyx/jots/NewJotTest.java  2
- core/src/main/java/com/larryhsiao/nyx/core/jots/PostedJot.java  2
- app/src/main/java/com/larryhsiao/nyx/jot/CalendarFragment.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotsByLocation.java  3
- core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java  2
- core/src/main/java/com/larryhsiao/nyx/core/tags/TagRemoval.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/JotWeekView.java  11
- core/src/main/java/com/larryhsiao/nyx/core/attachments/NewAttachments.java  2
- core/src/test/java/com/larryhsiao/nyx/jots/QueriedJotsTest.java  1
- core/src/main/java/com/larryhsiao/nyx/core/jots/QueriedJots.java  5
- core/src/main/java/com/larryhsiao/nyx/core/tags/QueriedTags.java  4
- core/src/main/java/com/larryhsiao/nyx/core/tags/TagDb.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/JotMapFragment.java  7
- core/src/main/java/com/larryhsiao/nyx/core/tags/JotTagRemoval.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/UpdateJot.java  3
- core/src/test/java/com/larryhsiao/nyx/jots/JotUriTest.java  1
- app/src/main/java/com/larryhsiao/nyx/tag/TagListFragment.java  3
- core/src/main/java/com/larryhsiao/nyx/core/attachments/ConstAttachment.java  1
- app/src/main/java/com/larryhsiao/nyx/base/JotActivity.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/ConstJot.java  1
- app/src/main/java/com/larryhsiao/nyx/jot/JotMonthView.java  11
- core/src/main/java/com/larryhsiao/nyx/core/tags/TagsByKeyword.java  3
- core/src/test/java/com/larryhsiao/nyx/jots/JotsByLocationTest.java  1
- core/src/main/java/com/larryhsiao/nyx/core/attachments/AttachmentDb.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/WrappedJot.java  1
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotsByIds.java  5
- app/src/main/java/com/larryhsiao/nyx/LocationString.java  7
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotsDb.java  3
- app/src/main/java/com/larryhsiao/nyx/base/JotFragment.java  1
- core/src/main/java/com/larryhsiao/nyx/core/tags/NewTag.java  5
         

Complexity decreasing per file
==============================
+ app/src/main/java/com/larryhsiao/nyx/JotApplication.java  -3
         

Clones added
============
- app/src/main/java/com/larryhsiao/nyx/jot/JotListFragment.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/JotWeekView.java  2
- app/src/main/java/com/larryhsiao/nyx/jot/JotMapFragment.java  3
- app/src/main/java/com/larryhsiao/nyx/tag/TagListFragment.java  2
- app/src/main/java/com/larryhsiao/nyx/jot/JotMonthView.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067&bid=16884172)

File: core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java (hunk 4 above)
Comment (LarryHsiao) [inline core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java]: Codacy Issue found: [Unused import - java.sql.DatabaseMetaData.](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067)

File: core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java (hunk 1 above)
Comment (LarryHsiao) [inline core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java]: Codacy Issue found: [Unused import - java.sql.ResultSet.](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067)

Comment (LarryHsiao) [inline core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java]: Codacy Issue found: [Avoid unused imports such as 'java.sql.ResultSet'](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067)

File: core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java (hunk 4 above)
Comment (LarryHsiao) [inline core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java]: Codacy Issue found: [Avoid unused imports such as 'java.sql.DatabaseMetaData'](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067)

**Comment by LarryHsiao** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- app/src/main/java/com/larryhsiao/nyx/base/JotActivity.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/CalendarFragment.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotRemoval.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/JotContentFragment.java  8
- core/src/main/java/com/larryhsiao/nyx/core/attachments/NewAttachment.java  5
- core/src/main/java/com/larryhsiao/nyx/core/jots/AllJots.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/AttachmentAdapter.java  1
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotsByIds.java  5
- app/src/main/java/com/larryhsiao/nyx/LocationString.java  7
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotsByDate.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotUriId.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/JotMapItem.java  1
- core/src/main/java/com/larryhsiao/nyx/core/tags/JotTagRemoval.java  3
- core/src/main/java/com/larryhsiao/nyx/core/attachments/ConstAttachment.java  1
- core/src/main/java/com/larryhsiao/nyx/core/tags/TagDb.java  3
- core/src/main/java/com/larryhsiao/nyx/core/tags/TagsByKeyword.java  3
- core/src/main/java/com/larryhsiao/nyx/core/tags/ConstTag.java  1
- core/src/main/java/com/larryhsiao/nyx/core/jots/UpdateJot.java  3
- core/src/main/java/com/larryhsiao/nyx/core/tags/NewTag.java  5
- app/src/main/java/com/larryhsiao/nyx/jot/JotListFragment.java  5
- core/src/test/java/com/larryhsiao/nyx/jots/JotUriTest.java  1
- app/src/main/java/com/larryhsiao/nyx/jot/JotMapFragment.java  7
- core/src/main/java/com/larryhsiao/nyx/core/jots/PointSource.java  2
- core/src/main/java/com/larryhsiao/nyx/core/jots/QueriedJots.java  5
- core/src/main/java/com/larryhsiao/nyx/core/tags/JotsByTagId.java  3
- core/src/test/java/com/larryhsiao/nyx/jots/JotsByLocationTest.java  1
- app/src/main/java/com/larryhsiao/nyx/jot/JotWeekView.java  11
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotsByLocation.java  3
- core/src/main/java/com/larryhsiao/nyx/core/tags/NewJotTag.java  3
- core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java  2
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotById.java  5
- app/src/main/java/com/larryhsiao/nyx/jot/JotMonthView.java  11
- core/src/main/java/com/larryhsiao/nyx/core/jots/NewJot.java  9
- app/src/main/java/com/larryhsiao/nyx/tag/TagListFragment.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/WrappedJot.java  1
- app/src/main/java/com/larryhsiao/nyx/tag/TagListAdapter.java  2
- app/src/main/java/com/larryhsiao/nyx/util/EmptyView.java  1
- core/src/main/java/com/larryhsiao/nyx/core/jots/PostedJot.java  2
- core/src/main/java/com/larryhsiao/nyx/core/tags/TagsByJotId.java  3
- core/src/main/java/com/larryhsiao/nyx/core/tags/TagRemoval.java  3
- core/src/main/java/com/larryhsiao/nyx/core/attachments/QueriedAttachments.java  4
- core/src/test/java/com/larryhsiao/nyx/jots/QueriedJotsTest.java  1
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotsByKeyword.java  3
- core/src/main/java/com/larryhsiao/nyx/core/attachments/NewAttachments.java  2
- core/src/test/java/com/larryhsiao/nyx/jots/JotUriIdTest.java  2
- app/src/main/java/com/larryhsiao/nyx/base/AuthActivity.java  1
- app/src/main/java/com/larryhsiao/nyx/jot/JotListAdapter.java  3
- core/src/test/java/com/larryhsiao/nyx/attachments/NewAttachmentTest.java  1
- core/src/main/java/com/larryhsiao/nyx/core/tags/AllTags.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/SampleJot.java  1
- core/src/main/java/com/larryhsiao/nyx/core/attachments/RemovalAttachmentByJotId.java  3
- app/src/main/java/com/larryhsiao/nyx/base/JotFragment.java  1
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotUri.java  1
- core/src/main/java/com/larryhsiao/nyx/core/tags/QueriedTags.java  4
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotsDb.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/ConstJot.java  1
- core/src/main/java/com/larryhsiao/nyx/core/attachments/AttachmentDb.java  3
- core/src/main/java/com/larryhsiao/nyx/core/attachments/AttachmentsByJotId.java  3
- core/src/test/java/com/larryhsiao/nyx/jots/NewJotTest.java  2
         

Complexity decreasing per file
==============================
+ app/src/main/java/com/larryhsiao/nyx/JotApplication.java  -3
         

Clones added
============
- app/src/main/java/com/larryhsiao/nyx/jot/JotListFragment.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/JotMapFragment.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/JotWeekView.java  2
- app/src/main/java/com/larryhsiao/nyx/jot/JotMonthView.java  2
- app/src/main/java/com/larryhsiao/nyx/tag/TagListFragment.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067&bid=16884172)

File: core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java (hunk 4 above)
Comment (LarryHsiao) [inline core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java]: Codacy Issue found: [Unused import - java.sql.DatabaseMetaData.](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067)

File: core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java (hunk 1 above)
Comment (LarryHsiao) [inline core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java]: Codacy Issue found: [Unused import - java.sql.ResultSet.](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067)

Comment (LarryHsiao) [inline core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java]: Codacy Issue found: [Avoid unused imports such as 'java.sql.ResultSet'](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067)

File: core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java (hunk 4 above)
Comment (LarryHsiao) [inline core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java]: Codacy Issue found: [Avoid unused imports such as 'java.sql.DatabaseMetaData'](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067)

**Comment by LarryHsiao** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- core/src/main/java/com/larryhsiao/nyx/core/jots/ConstJot.java  1
- app/src/main/java/com/larryhsiao/nyx/tag/TagListAdapter.java  2
- core/src/main/java/com/larryhsiao/nyx/core/tags/TagDb.java  3
- app/src/main/java/com/larryhsiao/nyx/base/AuthActivity.java  1
- app/src/main/java/com/larryhsiao/nyx/base/JotActivity.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotUri.java  1
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotsByDate.java  3
- core/src/test/java/com/larryhsiao/nyx/jots/JotUriTest.java  1
- core/src/test/java/com/larryhsiao/nyx/jots/JotUriIdTest.java  2
- app/src/main/java/com/larryhsiao/nyx/jot/AttachmentAdapter.java  1
- core/src/main/java/com/larryhsiao/nyx/core/tags/JotTagRemoval.java  3
- core/src/main/java/com/larryhsiao/nyx/core/tags/QueriedTags.java  4
- core/src/main/java/com/larryhsiao/nyx/core/attachments/AttachmentsByJotId.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/JotListFragment.java  5
- app/src/main/java/com/larryhsiao/nyx/jot/CalendarFragment.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/WrappedJot.java  1
- app/src/main/java/com/larryhsiao/nyx/jot/JotMapFragment.java  7
- core/src/main/java/com/larryhsiao/nyx/core/tags/TagsByKeyword.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/JotContentFragment.java  8
- core/src/main/java/com/larryhsiao/nyx/core/attachments/RemovalAttachmentByJotId.java  3
- core/src/main/java/com/larryhsiao/nyx/core/tags/JotsByTagId.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/QueriedJots.java  5
- core/src/main/java/com/larryhsiao/nyx/core/jots/SampleJot.java  1
- app/src/main/java/com/larryhsiao/nyx/jot/JotMonthView.java  11
- core/src/main/java/com/larryhsiao/nyx/core/attachments/NewAttachments.java  2
- core/src/main/java/com/larryhsiao/nyx/core/tags/AllTags.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotRemoval.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotUriId.java  3
- app/src/main/java/com/larryhsiao/nyx/LocationString.java  7
- core/src/main/java/com/larryhsiao/nyx/core/jots/NewJot.java  9
- core/src/test/java/com/larryhsiao/nyx/jots/QueriedJotsTest.java  1
- core/src/main/java/com/larryhsiao/nyx/core/tags/NewTag.java  5
- core/src/main/java/com/larryhsiao/nyx/core/jots/PostedJot.java  2
- core/src/main/java/com/larryhsiao/nyx/core/attachments/QueriedAttachments.java  4
- core/src/test/java/com/larryhsiao/nyx/jots/JotsByLocationTest.java  1
- app/src/main/java/com/larryhsiao/nyx/tag/TagListFragment.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/JotWeekView.java  11
- core/src/main/java/com/larryhsiao/nyx/core/jots/AllJots.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/UpdateJot.java  3
- core/src/test/java/com/larryhsiao/nyx/jots/NewJotTest.java  2
- core/src/main/java/com/larryhsiao/nyx/core/attachments/AttachmentDb.java  3
- core/src/test/java/com/larryhsiao/nyx/attachments/NewAttachmentTest.java  1
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotById.java  5
- core/src/main/java/com/larryhsiao/nyx/core/attachments/ConstAttachment.java  1
- core/src/main/java/com/larryhsiao/nyx/core/attachments/NewAttachment.java  5
- core/src/main/java/com/larryhsiao/nyx/core/tags/TagRemoval.java  3
- core/src/main/java/com/larryhsiao/nyx/core/tags/TagsByJotId.java  3
- core/src/main/java/com/larryhsiao/nyx/core/tags/ConstTag.java  1
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotsByLocation.java  3
- core/src/main/java/com/larryhsiao/nyx/core/NyxDb.java  2
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotsByIds.java  5
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotsByKeyword.java  3
- app/src/main/java/com/larryhsiao/nyx/base/JotFragment.java  1
- app/src/main/java/com/larryhsiao/nyx/jot/JotMapItem.java  1
- core/src/main/java/com/larryhsiao/nyx/core/jots/JotsDb.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/JotListAdapter.java  3
- core/src/main/java/com/larryhsiao/nyx/core/jots/PointSource.java  2
- app/src/main/java/com/larryhsiao/nyx/util/EmptyView.java  1
- core/src/main/java/com/larryhsiao/nyx/core/tags/NewJotTag.java  3
         

Complexity decreasing per file
==============================
+ app/src/main/java/com/larryhsiao/nyx/JotApplication.java  -3
         

Clones added
============
- app/src/main/java/com/larryhsiao/nyx/jot/JotListFragment.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/JotMapFragment.java  3
- app/src/main/java/com/larryhsiao/nyx/jot/JotMonthView.java  2
- app/src/main/java/com/larryhsiao/nyx/tag/TagListFragment.java  2
- app/src/main/java/com/larryhsiao/nyx/jot/JotWeekView.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5139067&bid=16884172)

**Comment by 0pdd** [comment]
@LarryHsiao the puzzle [#261](https://github.com/LarryHsiao/Nyx/issues/261) is still not solved.

**Comment by 0pdd** [comment]
@LarryHsiao the only puzzle [#261](https://github.com/LarryHsiao/Nyx/issues/261) is solved here.

**Pull request merged**